contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+     *                               tokens with the desired pre-configured shares
+     * 3. addLiquidity            -> add liquidity into the pool respecting share split configuration
+     *
+     * Scenario 1: When there is yet no liquidity yet in the pool (from the strategy or others) that means that
+     *             someone has minted the pool, added the initial liquidity and removed the liquidity from the
+     *             pool. (See `aerodrome_amo_liquidity.py` brownie script on block 18558804 to test the situation).
+     *             Then the Aerodrome pool is in this particular state where active trading price of the pool is
+     *             still at the value when last liquidity was in the pool and that trading price can not be moved
+     *             since there is no liquidity to perform the swap. In such a case the rebalancing transaction
+     *             shall be reverted.
+     *             Swap transaction is the one that shall fail. Unfortunately there is no easy way to query Aerodrome
+     *             slipstream pool for the amount of liquidity deposited. Even tokens on the contract can be from
+     *             fees, or liquidity that has been removed but not yet claimed.
+     *             It becomes the responsibility of the strategist or deployer to add some liquidity in the configured
+     *             tick ranges to the pool to facilitate the swap. Effectively turning Scenario 1 into a Scenario 2
+     * Scenario 2: When there is no liquidity in the pool from the strategy but there is from other LPs then
+     *             only step 1 is skipped. (It is important to note that liquidity needs to exist in the configured
+     *             strategy tick ranges in order for the swap to be possible) Step 3 mints new liquidity position
+     *             instead of adding to an existing one.
+     * Scenario 3: When there is strategy's liquidity in the pool all 3 steps are taken
+     *
+     *
+     * Exact _amountToSwap, _minTokenReceived & _swapWeth parameters shall be determined by simulating the
+     * transaction off-chain. The strategy checks that after the swap the share of the tokens is in the
+     * expected ranges.
+     *
+     * @param _amountToSwap The amount of the token to swap
+     * @param _swapWeth Swap using WETH when true, use OETHb when false
+     * @param _minTokenReceived Slippage check -> minimum amount of token expected in return
+     */
+    function rebalance(
+        uint256 _amountToSwap,
+        bool _swapWeth,
+        uint256 _minTokenReceived
+    ) external nonReentrant onlyGovernorOrStrategist {
+        _rebalance(_amountToSwap, _swapWeth, _minTokenReceived);
+    }
+
+    /**
+     * @dev Remove almost all of the liqudity, rebalance the pool to the desired token split and
+     * deposit all of the liquidity.
+     */
+    // rebalance already has reentrancy check
+    // slither-disable-start reentrancy-no-eth
+    function _rebalance(
```

Comment:
⚪️ We have this internal rebalance method with the same number parameters as the external method and it's only called by the external method. I think we can just move all this into the external method?

Reply:
Good point there is no need for a separate internal rebalance function: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/b1d9c1e2e3ae1cc0bea7156c2c6f73b564473b60